packages/legacy/core/App/hooks/chat-messages.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  BasicMessageRecord,
+  ConnectionRecord,
+  CredentialExchangeRecord,
+  CredentialState,
+  ProofExchangeRecord,
+  ProofState,
+} from '@credo-ts/core'
+import { useBasicMessagesByConnectionId } from '@credo-ts/react-hooks'
+import { isPresentationReceived } from '@hyperledger/aries-bifold-verifier'
+import { useNavigation } from '@react-navigation/core'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { Fragment, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Linking, Text } from 'react-native'
+
+import { ChatEvent } from '../components/chat/ChatEvent'
+import { ExtendedChatMessage, CallbackType } from '../components/chat/ChatMessage'
+import { useStore } from '../contexts/store'
+import { useTheme } from '../contexts/theme'
+import { useCredentialsByConnectionId } from '../hooks/credentials'
+import { useProofsByConnectionId } from '../hooks/proofs'
+import { Role } from '../types/chat'
+import { RootStackParams, ContactStackParams, Screens, Stacks } from '../types/navigators'
+import {
+  getConnectionName,
+  getCredentialEventLabel,
+  getCredentialEventRole,
+  getMessageEventRole,
+  getProofEventLabel,
+  getProofEventRole,
+} from '../utils/helpers'
+
+const callbackTypeForMessage = (record: CredentialExchangeRecord | ProofExchangeRecord) => {
+  if (
+    record instanceof CredentialExchangeRecord &&
+    (record.state === CredentialState.Done || record.state === CredentialState.OfferReceived)
+  ) {
+    return CallbackType.CredentialOffer
+  }
+
+  if (
+    (record instanceof ProofExchangeRecord && isPresentationReceived(record) && record.isVerified !== undefined) ||
+    record.state === ProofState.RequestReceived ||
+    (record.state === ProofState.Done && record.isVerified === undefined)
+  ) {
+    return CallbackType.ProofRequest
+  }
+
+  if (
+    record instanceof ProofExchangeRecord &&
+    (record.state === ProofState.PresentationSent || record.state === ProofState.Done)
+  ) {
+    return CallbackType.PresentationSent
+  }
+}
+
+export const useChatMessagesByConnection = (connection: ConnectionRecord): ExtendedChatMessage[] => {
```

Comment:
Is this fn doing more that just returning the messages for particular connection? If so it might be worth a JSDoc blurb explaining what is purpose is.